zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -28,38 +31,72 @@ import okio.IOException
 import okio.Path
 
 /**
- * Gets code from an HTTP server or a local cache,
- * and loads it into a zipline instance. This attempts
- * to load code as quickly as possible, and will
- * concurrently download and load code.
+ * Gets code from an HTTP server, or optional local cache
+ * or embedded filesystem, and loads it into a zipline instance.
+ * Loader attempts to load code as quickly as possible with
+ * concurrent network downloads and code loading.
  */
-class ZiplineLoader(
+class ZiplineLoader private constructor(
   private val dispatcher: CoroutineDispatcher,
   private val httpClient: ZiplineHttpClient,
-  private val embeddedDir: Path,
-  private val embeddedFileSystem: FileSystem,
-  cacheDbDriver: SqlDriver, // SqlDriver is already initialized to the platform and SQLite DB on disk
-  cacheDir: Path,
-  val cacheFileSystem: FileSystem,
-  cacheMaxSizeInBytes: Int = 100 * 1024 * 1024,
-  nowMs: () -> Long, // 100 MiB
+  private val fsCachingFetcher: FsCachingFetcher? = null,
+  private val fsEmbeddedFetcher: FsEmbeddedFetcher? = null,
 ) {
+  constructor(dispatcher: CoroutineDispatcher, httpClient: ZiplineHttpClient) : this(
```

Comment:
Shouldn't the HTTP client also be a fetcher?

Reply:
I was going back and forth on that one, I'll set it up as a fetcher too